### DISCUSSION TITLE
Performance on bulk/batch insert / update / delete

### DISCUSSION THREAD

**Discussion by leslie-qiwa** [go-pg/pg#1983] [General]
I'm wondering if anyone did performance analysis on bulk/batch insert / update / delete. Should it be recommended?

**Comment by elliotcourant** [comment]
Sorry for a late response here, but in general bulk will always be more performant than individual queries for one reason alone. Individual queries require a complete network round trip the DB each time, and then require that the DB examine and plan for each of these individual queries. Generally a bulk or batch query has the same cost in terms of network overhead as the sum individual queries but will likely be slightly better depending on your setup. But the DB will only need to create a single query plan for the bulk query, potentially saving time there.

A lot of this depends on your use case and your configuration, but if I were to make a blanket statement; batch/bulk queries will be more performant in an apples to apples comparison.
